### PR TITLE
Creates AI Assistant use cases section

### LINKED ISSUE #5233: [Request] Use-case based documentation [Using Elastic AI Assistant & Attack Discovery for Security & Incident Reporting]
### Description

What: Document how users can utilize Elastic AI Assistant & Attack Discovery for Security and Incident Report

### Background & resources

Docs draft: [link](https://docs.google.com/document/d/1Ib0hf1D3s4zMSZSsvNIu_6lwT7MvEMh7AdUR5sPoXzQ/edit?usp=sharing)

### Which documentation set does this change impact?

ESS and serverless

### ESS release

8.14

### Serverless release

tbd

### Feature differences

The feature is identical

### API docs impact

None

### Prerequisites, privileges, feature flags

None

### PR DESCRIPTION
**Pull request by benironside**
Closes #5233. 

This PR:
- Creates two new subsections within the AI Assistant section:
    - LLM connector guides. Combines three existing pages into a section, and adds a landing page for the section ([preview](https://security-docs_bk_5303.docs-preview.app.elstc.co/guide/en/security/master/llm-connector-guides.html)).
    - AI Assistant use-cases. Creates a new section from one existing and two new docs, and adds a landing page ([preview](https://security-docs_bk_5303.docs-preview.app.elstc.co/guide/en/security/master/assistant-use-cases.html)).
- Creates two new AI Assistant use-case guides:
    - Identify, investigate, and document threats with Elastic AI Assistant ([preview](https://security-docs_bk_5303.docs-preview.app.elstc.co/guide/en/security/master/attack-discovery-ai-assistant-incident-reporting.html)).
    - Generate, customize, and learn about {esql} queries with AI Assistant ([preview](https://security-docs_bk_5303.docs-preview.app.elstc.co/guide/en/security/master/esql-queries-assistant.html)).
- Makes minor edits to existing AI Assistant pages.
    - AI Assistant main page: adds a link to the LLM connector guides section, and updates the recommended models sidebar ([preview](https://security-docs_bk_5303.docs-preview.app.elstc.co/guide/en/security/master/security-assistant.html)).
    - Triage alerts with AI Assistant: just tweaked the title to match the new use-case guides.

### CODE REVIEW

**Comment by jamesspi** [comment]
@benironside , a few minor comments:

image

The above title is misleading, and should represent both the assistant and attack discovery. I'm thinking we can just drop the trailing "AI Assistant" from all these titles, and the security within the docs implies that specifically, unless you think it should be there for SEO?

image

Everything else looks ok to me.

Side note - For this page - https://security-docs_bk_5303.docs-preview.app.elstc.co/guide/en/security/master/attack-discovery.html - Can we embed this video, similar to what we did with the Alert triage video?

https://www.youtube.com/watch?v=xt1FR3AvuVg&t=1s

File: docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc
Comment (jmikell821) [inline docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:26]:
```suggestion
In the example above, Attack discovery found connections between eleven alerts, and used them to identify and describe an attack chain.
```

File: docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc
Comment (jmikell821) [inline docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:24]: Consider blurring sensitive information, like `james-spiteri` and the hostname.

File: docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc
Comment (jmikell821) [inline docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:28]:
```suggestion
After Attack discovery outlines your threat landscape, use Elastic AI Assistant to quickly analyze a threat in detail. 
```

File: docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc
Comment (jmikell821) [inline docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:34]:
```suggestion
From a discovery on the Attack discovery page, click **View in AI Assistant** to start a chat that includes the discovery as context. 
```

File: docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc
Comment (jmikell821) [inline docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:36]:
```suggestion
AI Assistant can quickly compile essential data and provide suggestions to help you generate an incident report and plan an effective response. You can ask it to provide relevant data or answer questions, such as “How can I remediate this threat?” or “What {esql} query would isolate actions taken by this user?” 
```

File: docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc
Comment (jmikell821) [inline docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:40]:
```suggestion
The image above shows an {esql} query generated by AI Assistant in response to a user prompt. Learn more about <<esql-queries-assistant,using AI Assistant for {esql}>>.
```

File: docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc
Comment (jmikell821) [inline docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:42]:
```suggestion
At any point in a conversation with AI Assistant, you can add data, narrative summaries, and other information from its responses to {elastic-sec}'s case management system to generate incident reports. 
```

File: docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc
Comment (jmikell821) [inline docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:48]:
```suggestion
From the AI Assistant dialog window, click **Add to case** (image:images/icon-add-to-case.png[Add to case icon,19,16]) next to a message to add the information in that message to a case. If you add a message that contains an Attack discovery to a case, AI Assistant automatically adds the attack summary and all associated alerts to the case. You can also add AI Assistant messages that contain remediation steps and relevant data to the case. 
```

File: docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc
Comment (jmikell821) [inline docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:55]:
```suggestion
AI Assistant can translate its findings into other human languages, helping to enable collaboration among global security teams and distributed Security Operations Centers (SOCs), and making it easier to operate within multilingual organizations. 
```

File: docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc
Comment (jmikell821) [inline docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:59]: What do we mean by "utility" here? 
```suggestion
NOTE: In our internal testing, AI Assistant translations preserved the accuracy and utility of the original content. 
```

File: docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc
Comment (jmikell821) [inline docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:63]: I'll let others weigh in on if they think this is necessary. Would be great as a sidebar component in our CSS styling, however :)

File: docs/assistant/ai-esql-queries.asciidoc
Comment (jmikell821) [inline docs/assistant/ai-esql-queries.asciidoc:15]:
```suggestion
* **Education and training**: AI Assistant can be a powerful {esql} learning tool. Ask it for examples, explanations of complex queries, and best practices.
```

File: docs/assistant/ai-esql-queries.asciidoc
Comment (jmikell821) [inline docs/assistant/ai-esql-queries.asciidoc:16]:
```suggestion
* **Writing new queries**: Prompt AI Assistant to provide a query that accomplishes a particular task, and it will generate a query that matches your description. For example: "Write a query to identify documents with `curl.exe` usage and calculate the sum of `destination.bytes`" or "What query would return all user logins to [a host] in the last six hours?"
```

File: docs/assistant/ai-esql-queries.asciidoc
Comment (jmikell821) [inline docs/assistant/ai-esql-queries.asciidoc:17]:
```suggestion
* **Providing feedback to optimize existing queries**: Provide AI Assistant a query you want to use and ask it for improvements, refactoring, a general assessment, or to optimize the query's performance with large data sets.
```

File: docs/assistant/llm-connector-guides.asciidoc
Comment (jmikell821) [inline docs/assistant/llm-connector-guides.asciidoc:4]:
```suggestion
This section contains instructions for setting up LLM connectors in {elastic-sec} so you can use <<security-assistant, Elastic AI Assistant>> and <<attack-discovery, Attack discovery>>. 
```

File: docs/assistant/ai-esql-queries.asciidoc
Comment (jmikell821) [inline docs/assistant/ai-esql-queries.asciidoc:21]:
```suggestion
In these ways and others, AI Assistant can enable you to make use of {esql}'s advanced search capabilities to accomplish goals within {elastic-sec}. 
```

**Review by jmikell821 — COMMENTED** [review]
A solid first draft. Just added some suggestions, which I hope you'll find useful.

File: docs/assistant/ai-alert-triage.asciidoc
Comment (benironside) [reply docs/assistant/ai-alert-triage.asciidoc:2]: Elastic AI Assistant is the full name, but guidance I've gotten was that AI Assistant is an acceptable shorthand.

File: docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc
Comment (benironside) [reply docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:24]: don't think this is necessary since James is so publicly involved with this feature. I see it as a little easter egg for those who've been following his work. Also don't think the demo hostname is really sensitive.

File: docs/assistant/assistant-use-cases.asciidoc
Comment (joepeeples) [inline docs/assistant/assistant-use-cases.asciidoc:8]: All these topics have great active, task-oriented titles!

File: docs/assistant/security-assistant.asciidoc
Comment (joepeeples) [inline docs/assistant/security-assistant.asciidoc:45]: Possible streamline:
```suggestion
You must create a generative AI connector before you can use AI Assistant. AI Assistant can connect to multiple large language model (LLM) providers so you can select the best model for your needs. To set up a connector, refer to <<llm-connector-guides,LLM connector setup guides>>.
```

File: docs/assistant/ai-esql-queries.asciidoc
Comment (joepeeples) [inline docs/assistant/ai-esql-queries.asciidoc:9]: It feels a little unnecessary for the page to announce itself in this way. (I know that works in other situations, so I'm not sure why it doesn't work for me here. Maybe because it's such a simple, direct statement, and the extra phrasing dilutes its effect?)
```suggestion
Elastic AI Assistant can help you learn about and leverage the Elasticsearch Query Language ({esql}). 
```

File: docs/assistant/ai-esql-queries.asciidoc
Comment (joepeeples) [inline docs/assistant/ai-esql-queries.asciidoc:25]: Hm, I really like how this "links richly" to build an EPPO-style information web 😉, but I'm not sure about the format. It's a super short section with just one sentence, which feels out of place (which don't typically have one-sentence sections in general). It might also feel more natural if ALL our pages all included a "Related docs" section, but it might not work as a one-off.

What about embedding these links into the intro paragraph instead? It already references "Elastic AI Assistant" and "Elasticsearch Query Language" so those would be good anchors.

File: docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc
Comment (joepeeples) [inline docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:16]: Seems like these tasks might not _need_ to be done in this specific order, so I'd go with bullets instead.

Comment (joepeeples) [reply docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:16]: Also holy cow AI Assistant can DO THIS? 😮 Amaze
> Translate incident information to a different human language using AI Assistant

File: docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc
Comment (joepeeples) [reply docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:24]: >  I see it as a little easter egg for those who've been following his work

I get that, but I'd still double-check to make sure he's OK with being easter-egged. 🍳 

Also the details in the image are VERY tiny, and then the next sentence directly references it and asks the reader to actually try to read the small text. It's a minor thing, so disregard this so if it'd take a lot of time, setup, mocking, etc. to retake.

File: docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc
Comment (joepeeples) [inline docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:34]: OK, now I'm seeing how maybe these sections DO go in a certain order, but I still think they could exist just as well as more standalone use cases. In this use case, we could suggest other UIs, workflows, etc. where a user can jump into AI Assistant to start analyzing — the Attack discovery page isn't the only jumping-off point.

File: docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc
Comment (joepeeples) [inline docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:50]: The second paragraph felt a little tacked on to me. Maybe:
```suggestion
From the AI Assistant dialog window, click **Add to case** (image:images/icon-add-to-case.png[Add to case icon,19,16]) next to a message to add the information in that message to a <<cases-overview,case>>. Cases help centralize relevant details in one place for easy sharing with stakeholders.

If you add a message that contains an Attack discovery to a case, AI Assistant automatically adds the attack summary and all associated alerts to the case. You can also add AI Assistant messages that contain remediation steps and relevant data to the case. 
```

File: docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc
Comment (joepeeples) [inline docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:55]: Nit: feels a little redundant (a distributed SOC is a global security team):
```suggestion
AI Assistant can translate its findings into other human languages, helping to enable collaboration among global security teams, and making it easier to operate within multilingual organizations. 
```

File: docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc
Comment (joepeeples) [inline docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:59]: Hm, I was actually expecting the opposite here: we're usually quite explicit about telling customers to take AI results with a grain of salt (for good reason!). Should we temper this note with the same healthy skepticism toward AI in general?

File: docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc
Comment (joepeeples) [reply docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:63]: I commented on the same type of section elsewhere (will link once I save this review).

**Review by joepeeples — COMMENTED** [review]
I left several comments but don't take that as criticism; excellent content here!

Semi-tangent: I'd prefer a different capitalization of "Attack discovery" — either "Attack Discovery" or "attack discovery" would flow better within sentences. But I know that's a different ball of worms, just figured I'd say it anyway. 😁

File: docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc
Comment (nastasha-solomon) [inline docs/assistant/use-attack-discovery-ai-assistant-incident-reporting.asciidoc:50]: Not sure if this edit is correct, so feel free to reject it if it's not.

```suggestion
If you add a message that contains a discovery to a case, AI Assistant automatically adds the attack summary and all associated alerts to the case. You can also add AI Assistant messages that contain remediation steps and relevant data to the case. 
```

**Review by nastasha-solomon — DISMISSED** [review]
Let one tiny edit for your consideration. Otherwise, lgtm! 

Amazing job writing this up! 👏🏼👏🏼👏🏼

**Review by joepeeples — APPROVED** [review]

**Pull request merged**
